.vscode/launch.json

```diff
@@ -0,0 +1,75 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            // Use IntelliSense to find out which attributes exist for C# debugging
+            // Use hover for the description of the existing attributes
+            // For further information visit https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger-launchjson.md
+            "name": "Run Nitrox.Launcher",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "build-launcher",
+            // If you have changed target frameworks, make sure to update the pro gram path.
```

Comment:
Since nobody on our team uses VSCode at the moment, it would be good to document where this template was copied from. Maybe even add a page to the Nitrox wiki later.

Reply:
The team has decided that we prefer not include .vscode files until there are more people using it. This because it will likely break and we don't have any maintainers at this time to update it. You mind if you add the .vscode folder to .gitignore for now until it gets more use?

Reply:
I don't mind. However, there's very little that can actually break in the .vscode files, since they essentially do two things:
1. Run `dotnet build` on the project 
2. Instruct Microsoft's .NET debugger where the dll is
The .NET debugger then launches the dll with the debugger attached. Unless Microsoft changes the debuggers name from "coreclr" to something else, or changes how the JSON works, nothing can break. 
The JSON files are generated from the C# extension's template, which I've just copy-and-pasted for all the Nitrox projects. 
To try it out, delete the .vscode folder and from the sidebar click the "Debug" button and it will prompt you to create a launch configuration, which it will then auto fill.

Reply:
And also, vscode is basically the only viable C# editor, since VS doesn't exist, and JB Rider costs money. Not to mention most linux users prefer open-source free software.

Reply:
I agree with your reasoning. We did a rethink, but still slightly prefer more to not have it. This could easily change in the future, or there could be alternatives (reusing files from other IDEs like .editorconfig).

Reply:
VSCode's C# extension can read .editorconfig, but it needs a setting to enable it. This setting should be stored at workspace-level .vscode. 

Note that VSCode by default also doesn't enable analyzers, and needs a setting to enable it. I've commited some changes that allow for these to happen.

If these settings don't exist, anyone using VSCode will commit code that isn't formatted according to `CONTRIBUTING.md`, and doesn't use analyzers, leaving the user guessing at what is correct formatting.

Reply:
Let's keep it out of this PR for now. I'll add you to the team and you can discuss it with the team yourself. Right now the consensus is no.

Reply:
Seems good. Only thing left to review is the new key-value store.